sway-core/src/parse_tree/call_path.rs

```diff
@@ -24,19 +25,25 @@ impl std::convert::From<Ident> for CallPath {
     }
 }
 
-impl fmt::Display for CallPath {
+impl<T> fmt::Display for CallPath<T>
+where
+    T: fmt::Display,
+{
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         let mut buf = String::new();
         for prefix in self.prefixes.iter() {
             buf.push_str(prefix.as_str());
             buf.push_str("::");
         }
-        buf.push_str(self.suffix.as_str());
+        buf.push_str(&self.suffix.to_string());
```

Comment:
**_Hot tip_**: Not super important, but if `suffix` is a `T` where `T: Display`, you can do:

```rust
write!(&mut buf, "{}", self.suffix);
```
which avoids the extra `String` alloc from `to_string()`.

Alternatively, this whole `impl` should probably write directly into the `Formatter` with `write!(f, ...)`, but a change like that could probably be left for a separate PR.